Parsers/ASimWebSession/Parsers/vimWebSessionF5ASM.yaml

```diff
@@ -0,0 +1,261 @@
+                                    "SrcIpAddr",
+                                    temp_DstMatch1,
+                                    "DstIpAddr",
+                                    "No match"
+                                )
+          | where ASimMatchingIpAddr != "No match"
+          | parse-kv DeviceCustomString3 as (Host: string, ["User-Agent"]: string, Cookie: string, Referer: string) with (pair_delimiter="\\r\\n", kv_delimiter=":")
+          | parse DeviceCustomString3 with * "HTTP/" HttpVersion: string "\\r\\n" rest: string
+          | extend
+              EventResultDetails = tostring(FieldDeviceCustomNumber1)
+          | extend
+              EventResult = iff(toint(EventResultDetails) >= 400 or DeviceAction =~ "blocked", "Failure", "Success")
+          | where eventresult == '*' or EventResult =~ eventresult
+          | project-rename 
+              DstIpAddr = DestinationIP,
+              DstPortNumber = DestinationPort,
+              EventOriginalUid = ExtID,
+              HttpRequestMethod = RequestMethod,
+              NetworkApplicationProtocol = ApplicationProtocol,
+              HttpCookie = Cookie,
+              HttpHost = Host,
+              HttpReferrer = Referer,
+              HttpUserAgent = ['User-Agent'],
+              HttpRequestXff = DeviceCustomString5
+          | extend
+              HttpStatusCode = EventResultDetails,
+              AdditionalFields = bag_pack(
+                        "Full Request", DeviceCustomString3, 
+                        "Policy Name", DeviceCustomString1,
+                        "Attack Type", DeviceCustomString4,
+                        "Policy Apply Date", DeviceCustomDate1
+                    );
+      let AnomalyDetectionData = AllData
+          | where DeviceEventClassID in (DeviceEventClassIDList)
+          | where array_length(httpuseragent_has_any) == 0 
+          | where array_length(eventresultdetails_in) == 0
+          | extend temp_DstMatch2 = has_any_ipv4_prefix(DvcIpAddr, ipaddr_has_any_prefix)
+          | extend ASimMatchingIpAddr = case(
+                                    array_length(src_or_any) == 0,
+                                    "-",
+                                    temp_SrcMatch and temp_DstMatch2,
+                                    "Both",
+                                    temp_SrcMatch,
+                                    "SrcIpAddr",
+                                    temp_DstMatch2,
+                                    "DstIpAddr",
+                                    "No match"
+                                ),
+                EventResult = iff(DeviceAction =~ "passed", "Success", "Failure")
+          | where ASimMatchingIpAddr != "No match"
+          | where eventresult == '*' or EventResult =~ eventresult
+          | extend
+              DstIpAddr = DvcIpAddr,
+              DstHostname = DvcHostname,
+              EventOriginalUid = tostring(FieldDeviceCustomNumber3),
```

Comment:
FieldDeviceCustomNumber3 is value for 'attack_id'. Seems more relevant for ThreatId

Reply:
@niralishah-crest 
Please map only for Anomaly related events. My understanding is that Anomaly events contain threat info. Is this understanding correct?

Reply:
Yes. Updated.